summary/buffer.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package summary
+
+import (
+	"github.com/google/cadvisor/info"
+)
+
+// Manages a buffer of usage samples.
+// This is similar to stats buffer in storage/memory.
+// The main difference is that we do not pre-allocate the buffer as most containers
+// won't live that long.
+type SamplesBuffer struct {
+	// list of collected samples.
+	samples []info.Usage
```

Comment:
I was planning on using this in heapster :)

On Fri, Feb 6, 2015 at 10:35 AM, Victor Marmol <notifications@github.com>
wrote:

> In summary/buffer.go
> <https://github.com/google/cadvisor/pull/491#discussion_r24260796>:
>
> > +// See the License for the specific language governing permissions and
> > +// limitations under the License.
> > +
> > +package summary
> > +
> > +import (
> > +	"github.com/google/cadvisor/info"
> > +)
> > +
> > +// Manages a buffer of usage samples.
> > +// This is similar to stats buffer in storage/memory.
> > +// The main difference is that we do not pre-allocate the buffer as most containers
> > +// won't live that long.
> > +type SamplesBuffer struct {
> > +	// list of collected samples.
> > +	samples []info.Usage
>
> It's probably fine to keep it stats-specific until we have another user.
> We save from all the type casting :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/cadvisor/pull/491/files#r24260796>.
>